drizzlepac/haputils/cell_utils.py

```diff
@@ -648,6 +812,375 @@ def build_mask(self):
+        self.inside = trace_polygon(inside - ndimage.binary_erosion(inside).astype(np.int16))
+
+
+        # Start by identifying what portions of each chip edge makes up
+        # the exterior of the total footprint as specified on input
+        for edge in self.edges:
+            segments, sky_segments = find_segments(edge['edge'][0], edge['edge'][1], self.inside, wcs)
+            if len(segments) > 0:
+                self.segments.append({'edge': segments, 'sky_edge': sky_segments})
+
+        # Now connect all exterior chip segments into a continuous polygon
+        # Start with the corner (start/end of edge) with largest Y value
+        start_corner = None
+        region = []
+        # define what edge will close/finish the polygon when going counter-clockwise
+        # This edge defines the last 2 points on the sky for the S_REGION keyword
+        last_edge = None
+        start_edge = None
+        remaining_segments = 0
+
+        for segment in self.segments:
+            # The segment may contain 0 or multiple segments
+            for start, end in segment['sky_edge']:
+                remaining_segments += 1
+                # Look for starting corner as corner with largest Dec (closest to +90)
+                # which has the smallest RA (if multiple points have identical Dec)
+                if (start_corner is not None and ((start[1] > start_corner[1]) or \
+                    start[1] == start_corner[1] and start[0] < start_corner[0])) or \
+                    start_corner is None:
+                    start_corner = start
+                    start_edge = [start, end]
+
+                if end[1] > start_corner[1]:
+                    start_corner = end
+                    last_edge = [start, end]
+                    start_edge = None  # Try again to find starting edge
+
+        # After finding starting corner, if last_edge is not None, we know
+        # that the starting corner is the 'start' of the segment when
+        # going counter-clockwise
+        end_corner = None
+        if start_edge is None:
+            # last_edge will always be defined in this case, so
+            # look for edge with start closest to this edge end point
+            for edge in self.segments:
+                for start, end in self.segments['sky_edge']:
+                    if start == last_edge['sky_edge'][1]:
+                        start_edge = edge
+                        end_corner = end
+                        break
+        else:
+            # start populating the polygon (s_region coordinates)
+            end_corner = start_edge['sky_edge'][1]
+
+        # At this point, both the starting edge and end corner are defined.
+        # Now start building up region polygon from all exterior segments
+        # starting from this edge
+        remaining_segments -= 1
+        region.append(start_edge)
+        region_end = None
+
+        while region_end != end_corner and remaining_segments > 0:
```

Comment:
I guess I am sleepy, but what is the purpose of this while loop?

Reply:
This whole class ended up being un-used and so it was deleted in 07d2e887.